---
title: "Mutation rate variability in cetacean genomes"
author: "Amy M. Van Cise^1\\*â€ ^, Sophia Garrote^1^, Annabel Beichman^2^, Phillip A. Morin^3^"


output: 
    html_document:
      fig_caption: yes
      
bibliography: "`r rbbt::bbt_write_bib('bibliography.json', overwrite = TRUE)`"

knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_dir = "docs",
  output_file = "index.html") })
---

1. School of Aquatic and Fishery Sciences, University of Washington, Seattle, WA
3. Genome Sciences, University of Washington, Seattle, WA
2. Southwest Fisheries Science Center, National Marine Fisheries Service, National Oceanic and Atmospheric Administration, La Jolla, CA

<br>

^\*^Corresponding author email: avancise\@gmail.com

<br>

Running page head: Genomic mutation rate in cetaceans

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)

library(tidyverse)
library(ggsci)
library(PNWColors)
library(pander)
library(patchwork)
```

```{r data, cache = FALSE, warning = FALSE, message = FALSE, results = 'hide'}

load("Mutation-rate-coding-noncoding.Rdata")
load("whole_mutation_rate_df.Rdata")

```


<br>

## Abstract {-}

Genomic mutation is the fundamental process driving evolutionary processes that result in contemporary species, subspecies, and populations. Thus, genomic studies of divergence, speciation, and evolutionary demographics rely on accurate estimates of mutation rate. Historically, mutation rates have been estimated using a small portion of the genome and/or averaged among multiple species within a family or order, but growing evidence indicates potentially high variability in mutation rate both across the genome and among species. Here, we examine variability in mutation rate within cetaceans using high-quality genomes from the Vertebrate Genome Project. We estimate mutation rate across a range of parameters, including choice of input reference genome, coding vs. non-coding regions, chromosome-specific, species-specific, and using a variety of read depths, divergence date estimate, and methods to calculate mutation rate, in order to determine which factors have the greatest effect on variability in mutation rate. We find that mutation rate is most variable among species, and consequently most affected by the choice of reference genome used in the alignment pipeline. Other parameters had trivial effects on mutation rate estimates in comparison. These results indicate the importance of thoughtful selection of the reference genome used in future cetacean genomic studies, in order to ensure accurate downstream analyses of evolutionary processes.

## Introduction {-}

As the fundamental process underlying evolution, mutation drives processes such as divergence among species and evolutionary adaptation. The rate at which mutations accrue will ultimately determine the rate at which species diverge from each other or adapt to changes in their environment. 

Contemporary estimates of species divergence and adaptation that are central to evolutionary and population genetics rely on accurate estimates of mutation rate. Biases in this estimate may be reflected in many downstream analyses that shape our understanding of evolutionary biology, e.g. the timing of splits in genetic phylogenies, patterns of selective divergence as they relate to neutral variation, or the accuracy of evolutionary demographic analyses. They may also have affect conservation priorities and management actions. For example, a recent analysis of baleen whale mutation rates based wild pedigrees indicates that mutation rates in these species is higher than previously estimated based on phylogenetic divergence; correcting the mitochondrial mutation rate previously used in genetic estimates of pre-exploitation North Atlantic humpback whale abundance reduces those estimates by 86% [@suarez-menendez_etal23] and considerably shifts our understanding of the recovery of that population. 

Evolutionary demographic analysis relate genetic diversity, or the number of mutations in a given genetic lineage, to abundance via the coalescent theory. In these types of analyses especially, the mutation rate estimate used may have a drastic impact on our understanding of historic abundance of a population - and changes in the mutation rate over time could also affect the observed trends in abundance over time. Early studies of mutation rate suggest a lack of direct relationship between mtDNA genetic diversity and population size [@nabholz_etal08, REFS 18 from Nabholz]. Further, a growing body of evidence indicates that mutation rate varies across the genome, and that site-specific variability is only partially context-dependent [@ellegren_etal03; @hodgkinson_eyre-walker11]. Finally, mutation rate has been shown to vary over evolutionary timescales (REF).

In addition to varying throughout the genome and through time, mutation rates have been shown to vary widely across species [e.g. @Nabholz2009a; @bergeron_etal23; REFS 14, 43, 39 from Nabholz]. Three broad hypotheses have been proposed to explain potential drivers of mutation rate variability among species. (1) The longevity hypothesis posits that mutation rate decreases with increased lifespan, allowing individuals to live for longer periods of time without adverse effects of the accumulation of harmful mutations over their lifetimes. (2) The generation time hypothesis, which posits that mutation rate XX. (3) The metabolic rate hypothesis posits that mutation rate increases with increasing metabolic rate (often proxied by body mass) due to the oxidative effects of metabolism on the cell.

The effects of demography, phylogeny, and genome location on mutation rate highlight the need to quantify mutation rate variability within taxonomic groups in order to inform downstream evolutionary and conservation genetic analyses. In addition to these factors, there are multiple decision points along the bioinformatic pipeline from sequence data to aligned genome that may affect the accurate tally of genomic mutations and downstream estimation of mutation rate. However, there is limited research on how the selection of these input parameters affects mutation rate estimates.

The increasing accessibility of genomic sequencing technologies and high-quality reference genomes for marine mammals (cite VGP here) has precipitated an rise in the use of genomic techniques to address previously unanswered questions about the ecology and conservation of these species. Because these species are rare, remote, elusive, and protected, it is often difficult to obtain the sample size needed to accurately detect and describe ecological processes and patterns. In cases such as this, whole genome analyses increase sensitivity and power to describe genetic patterns using a smaller sample set, thus making it a potentially powerful tool for studies of evolutionary ecology in marine mammals. These genomic studies in marine mammals, e.g. descriptions of evolutionary phylogenies or historical demographics of species, currently rely on estimates of mutation rate that are averaged across a family of species and often based on only a small portion of the genome. 

Here, we explore mutation rate variability within the cetaceans, and describe the major drivers affecting variability in estimates of mutation rate, with the aim of providing insight and context into the estimation of an accurate and appropriate mutation rate in cetacean genomics studies. We examine process driven variability by estimating taxon- and chromosome-specific mutation rates, and comparing species-specific mutation rates with key demographic traits considered to affect mutation rate. We also examine potential drivers of observation error by varying key parameters in the pipeline used to align genomes and estimate a mutation rate. These input parameters include the choice of reference genome used for alignment, minimum and maximum read depth threshold for genotyping a locus, inclusion of coding or non-coding regions in the mutation rate estimate, divergence date estimate used in the mutation rate calculation, and whether or not ancestral heterozygosity is included in the mutation rate calculation.

## Methods {-}

### Sample and reference genome preparation {-}

We acquired genomes from fourteen cetacean species, representing the breadth of cetacean phylogenetic diversity, from NCBI Genbank and the European Nucleotide Archive using SRAToolkit (REF, Table 1). Genomic data were generated by the Vertebrate Genome Progect (REF) using either Illumina short-read or Dovetail Hi-C approaches (REFS?). Prior to sequence assembly, we formatted all sequence files using BBMap (REF) to flag forward and reverse reads. Sequence files were formatted  assembly using BBMap (REF). 

In order to test for the effect of reference genome on the resulting mutation rate, we selected four reference genomes representing a deeply diverged taxon, a recently diverged taxon, and an intermediate taxon (Table 2). The deeply diverged taxon is represented by the hippopotamus (*Hippopotamus amphibius*), and the intermediate taxon is represented by the sperm whale (*Physeter macrocephaluS*). We selected two species to represent the recently diverged taxon: the killer whale (*Orcinus orca*) is used for all odontocetes, and the North Atlantic right whale (*Eubalaena glacialis*) is used for all mysticetes.

We downloaded all reference genomes as fastq files from NCBI Genbank, and created index files using bwa (REF). Sample genomes that were generated using Dovetail Hi-C require additional index files, which we generated using bwa-mem2 and samtools (REFS). Finally, we mapped repeats and created a repeat masker for each reference file using bedtools (REF). All steps described here, making up the pre-alignment pipeline, are outlined at: https://github.com/UW-WADE-lab/Whole-Genome-Alignment

```{r sample_info_table}

sample_table_format <- sample_table_data %>% 
  rename("Species" = species, "Abbreviation" = abbreviation, "Accession" = "accession..", "Recent reference" = close.reference) %>% 
  mutate(Database = c(rep("NCBI", 6),"ENA",rep("NCBI", 4), rep("ENA",3))) %>% 
  relocate(Database, .before = Accession)

pander(sample_table_format, split.table = Inf, table.caption.prefix = "Table 1:", caption = "Species abbreviation codes, genome source database, and accession number for each species included in the present study. For all species included in the multiple reference comparison, the final column indicates which species was useds as the recently diverged reference. **Add columns with generation time and reference for each species**")

pander(reference_table_data, split.table = Inf, table.caption.prefix = "Table 2:", caption = "Species abbreviation codes, genome source database, and accession number for each species used as a reference genome for alignment in this study.")

```

<br>

<br>

### Sample alignment {-}

All 14 sample genomes were first aligned to the intermediate reference genome, the sperm whale. The alignment pipeline, regardless of sequencing approach, comprised six steps: 1) align sample genome to reference, 2) sort aligned genome, 3) remove duplicate sequences, 4) mask repeat regions, 5) index genome, 6) call genotypes and filter variants by read depth. All alignment scripts can be found at https://github.com/UW-WADE-lab/Whole-Genome-Alignment.

Step 1 differed for Illumina short-read and Dovetail Hi-C sample genomes. We aligned forward and reverse reads from Illumina short-read genomes using bwa (REF). Dovetail Hi-C genomes, which comprise longer reads that are vulnerable to chimeras resulting from hairpin turns (?) (REF), require an additional filtering step post-alignment. Therefore, we aligned Dovetail Hi-C forward and reverse reads separately using bwa-mem2, and filtered forward and reverse read files separately using a custom perl script generated by Arima (REF/website here?) to identify and remove chimeras. We then combined forward and reverse reads using a custom perl script from Arima (REF/website here?).

Steps 2-6 were identical for both types of sample genome. We sorted genomes using samtools (REF) and removed duplicates using Picard (REF). We masked repeat regions, previously identified using the reference genome, with bedtools (REF). We indexed the sample genome using samtools, called genotypes using bcftools (REF), and filtered variants to remove those with a read depth greater than 500, also using bcftools. Note that this read depth filter is higher than is normally used in an alignment pipeline; this allowed us to directly examine the effect of read depth on mutation rate later in the study. 

We selected seven of the 14 sample genomes for a comparison of mutation rate estimates among reference genomes. These samples were additionally aligned to the hippopotamus and either the killer whale (odontocetes) or the North Atlantic right whale (mysticetes) using the pipeline described above (Table 1).

### Data analysis: calculating mutation rate {-}

Mutation rate ($\mu$) is commonly estimated as genomic divergence (i.e. genomic heterozygosity, $d_{xy}$) over divergence time ($t$):

\begin{equation}
    \mu=d_{xy}/2t
\end{equation}

Genomic divergence ($d_{xy}$), in turn, is estimated as:

\begin{equation}
    d_{xy}=(h_1+2h_2)/2g
\end{equation}

where $h_1$ is the number of sites in which the sample genome is heterozygous, $h_2$ is the number of sites in which the sample genome is homozygous for the alternate allele, and $g$ is the number of sites in which both the sample and reference genomes had called genotypes.

This calculation may result in an overestimate of mutation rate because ancestral (pre-divergence) heterozygosity is not taken into account, which we can account for in the following:

\begin{equation}
    \mu=(d_{xy}-\pi_{anc})/2t
\end{equation}

Because $\pi_{anc}$ is unknown, we used previous estimates of genome-wide heterozygosity from the harbor porpoise, as in Robinson et al. [-@Robinson2022].

Both divergence time ($t$) and ancestral heterozygosity $\pi_{anc}$ have previously been identified as potential sources of variability in downstream estimates of mutation rate. Therefore, we began by comparing estimates of mutation rate with and without ancestral heterozygosity, and using divergence times from McGowen et al. [-@mcgowen_etal20], XXXX., and TimeTree [@kumar_etal17]. We found that mutation rate variability owing to these two parameters was not significant (**TEST THIS, p-value here**) and trivial in comparison to variability caused by other parameters tested in this study (Supplemental Figure S1); therefore, all downstream data analyses used mutation rate estimates without ancestral heterozygosity and with divergence times from McGowen et al. [-@mcgowen_etal20].

The above equations produce a per-site mutation rate estimate, which we further converted to an estimate of mutation rate per site per generation. For most species, generation time was available from Taylor et al. [-@Taylor2007, Table 1].

### Data analysis: phylogenetic or demographic effects on mutation rate {-}

All data analyses were completed in R [@RCoreTeam2016] using Rstudio (REF). All 14 species are included in analyses oh phylogenetic and demographic effects on mutation rate, aligned to the sperm whale reference genome. Using a Shapiro-Wilk normality test [@royston82; @royston82a], we determined that our mutation rate dataset including 14 species aligned to the sperm whale reference are normally distributed (p = `r round(rate_normality$p.value, digits = 2)`), therefore we used ANOVA [@hollander_etal15p] for all tests of mutation rate differentiation among categorical taxonomic strata, and linear regression to test for demographic effects.

To test for an effect of phylogenetic inertia, we test for significant differentiation in whole-genome mutation rate between infraorders and among three families for which we have more than one species. We also consider potential effects of lifespan and body size on whole-genome mutation rate using data on species-specific maximum lifespan from XXXX (REF) and maximum body length from XXX (REF). Because generation time is used in our mutation rate estimates, we do not test for a relationship between this parameter and mutation rate. We first tested for correlation between lifespan and maximum body length and found them to be highly correlated (VALUE HERE); therefore, our linear model included only maximum lifespan as a predictor of whole-genome mutation rate. We tested the distribution of our mutation rate data for normality in order to confirm that a traditional linear relationship could appropriately be used to model our data, and further confirmed model fit by comparing R^2^ values among several models and by visually examining model residuals. Finally, because phylogeny may be a significant driver of mutation rate among species, we compared our linear model with a linear mixed model, setting family as a random effect.

### Data analysis: pipeline effects on mutation rate {-}

Within this study, we consider the effect of the following pipeline variability on downstream estimates of mutation rate:

1. Choice of reference species for genomic alignment
2. Genomic location of regions used in estimate (i.e. chromosome-level mutation rate variability)
3. Inclusion of coding vs. non-coding regions in estimate
4. Maximum read depth cutoff for called genotypes

**Reference species:** As described above, we additionally aligned seven of the fourteen species to a deeply diverged reference (hippopotamus) and a recently diverged reference (North Atlantic right whale or killer whale, see Table 2). We when grouped whole genome mutation rates into three categories: deep divergence, intermediate divergence, and recent divergence, and used an ANOVA analysis to determine whether mutation rate varies significantly among these three groups.

**Chromosome position:** For all fourteen species aligned to the sperm whale reference genome, we extracted SNPs located on each chromosome using a custom bash script and estimated a chromosome-specific mutation rate for each species. We then used a chi-square analysis to determine whether mutation rate varies significantly among chromosomes across all species included in the study.

**Coding vs non-coding regions:** For all fourteen species aligned to the sperm whale reference genome, we extracted SNPs located in coding and non-coding regions by mapping to a bed file generated from annotations of the sperm whale reference genome (available from NCBI Genbank). We then compared mutation rate estimates between coding and non-coding regions of the genome, first across all fourteen species included in the study, and then within the mysticete and the odontocetes.

**Genotype read-depth:** In order to examine effects of genotype read-depth on variabilily in mutation rate estimates, we first generated new genotype vcf files with calls for all loci, including both heterozygous and homozygous loci. Using these vcf files and considering read depths of 10-500 reads, we then used a custom R script to determine the cumulative proportion of heterozygous sites with increasing read depth. Following this, we calculated mutation rate for each species at maximum read depth cutoffs from 100-500. Finally, using a generalized linear mixed model with species as a random effect, we test whether read depth is a significant driver of mutation rate estimates.


## Results {-}

### Phylogenetic or demographic effects on mutation rate

Whole genome mutation rate variability is high among individual species (Figure 1). We found that infraorder accounted for approximately 34% of this variability, with mutation rates that were significantly higher in mysticetes than in odontocetes  (ANOVA p = `r format(summary(infraorder_anova)[[1]][1,5], digits = 2)`, sum-of-square ratio = `r format(summary(infraorder_anova)[[1]][1,2]/(summary(infraorder_anova)[[1]][1,2]+summary(infraorder_anova)[[1]][2,2]), digits = 2)`, Figure 1). However, we did not observe a significant difference in mutation rate among the three families with X or more species included in this study: Balaenopteridae, Delphinidae, and Ziphiidae (ANOVA p = `r format(summary(family_anova)[[1]][1,5], digits = 2)`).

<br>

<br>

```{r whole_genome_rate, fig.align = "center", fig.height = 8, fig.width = 8, fig.cap = "Figure 1. Top: Whole genome mutation rate for each cetacean species included in the study. Bottom: Mean mutation rate for mysticete and odontocete clades. Mutation rates in both subplots were estimated using Pmac as the reference genome and divergence estimates from McGowen et al. 2020. The dotted line represents an odontocete nuclear mutation rate estimate from Dornburg et al. 2012."}
top <- ggplot(data=species_mrate_gen, aes(x=abbrev,y=rate)) +
  geom_point(size=3.7) +
  labs(x="Species", y="Mutations/\nsite/generation") +
  theme_light() +
  theme(text = element_text(size = 16), plot.subtitle = element_text(size = 14)) +
  guides(color=FALSE) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0))) +
  scale_shape_discrete(guide="none") +
  geom_hline(aes(yintercept=1.08e-8, linetype="dotdash")) +
  scale_linetype_discrete(guide="none")


bottom <- ggplot(data=species_mrate_gen, aes(x=infraorder, y=rate, fill=infraorder, color=infraorder, alpha=0.6)) +
  geom_violin() +
  labs(x="Infraorder", y="Mutations/\nsite/generation") +
  theme_light() +
  theme(text = element_text(size = 16)) +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_alpha(guide="none") +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none") +
  geom_dotplot(binaxis = 'y', size=4, fill = "black", color ="black", alpha = 1)
  
top / bottom
```

<br>

<br>

We gathered data on three demographic parameters with the potential to affect species-specific mutation rate: maximum lifespan, maximum body length (as a proxy for metabolic rate), and generation time. We first tested the degree of correlation between these three parameters (Supplemental Figure S2). Because body mass and lifespan are highly correlated (correlation = 0.84), we used only lifespan in our linear regression.

### Pipeline effects on mutation rate

The reference genome used affected mutation rate variability by an order of magnitude in some species (Figure X). Reference genomes for this analysis were chosen to reflect three periods of evolutionary divergence: deep, medium, and shallow. Mutation rate was highest in most species when compared to the deep divergence reference genome, indicating that mutation rate in cetaceans may have been higher in deep evolutionary time than current evolutionary time. Alignment to the middle divergence reference genome resulted in the lowest mutation rate estimates, and alignment to the shallow divergence reference genome resulted in mutation rate estimates that were intermediate between the deep and middle divergence reference genomes.

<br>

<br>

```{r multi_refs,  fig.align = "center", fig.height = 8, fig.width = 8, fig.cap = "Figure 2. Top: Mutation rate using different reference genomes with various levels of evolutionary distance for each sample with multiple references. The dotted line represents an odontocete nuclear mutation rate estimate from Dornburg et al. 2012. Bottom: Mutation references for each of the references used, Egla is the close reference for mysticetes and Oorc is the close reference for odontocete. Mutation rates in both subplots were estaimted using divergence estimates from McGowen et al. 2020."}

top <- ggplot(data=mult_refs_gen, aes(x=abbrev,y=rate,shape=div_est,color=ref_dist)) +
  geom_point(size=3.7) +
  labs(x="Species", y="Mutations/\nsite/generation") +
  theme_light() +
  theme(text = element_text(size = 16), plot.subtitle = element_text(size = 14)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none") +
  scale_shape_discrete(guide="none") +
  scale_color_manual(values=pnw_palette(n=3,name="Sunset2"), 
                     name="Reference Distance") +
  geom_hline(aes(yintercept=1.08e-8, linetype="dotdash"))
  
  
bottom <- mult_refs_gen %>% 
  arrange(rate) %>%
  mutate(reference = factor(reference, levels=c("Egla", "Oorc", "Pmac", "Hamb"))) %>%
  mutate(ref_dist = factor(ref_dist, levels=c("Close", "Middle", "Far"))) %>%
  ggplot(aes(x=reference, y=rate, fill = ref_dist, color = ref_dist, alpha = 0.6)) +
  geom_violin() +
  labs(x="Reference genome", y="Mutations/\nsite/generation") +
  scale_fill_manual(values=pnw_palette(n=3,name="Sunset2"), 
                     name="Reference Distance") +
  scale_color_manual(values=pnw_palette(n=3,name="Sunset2"), 
                     name="Reference Distance") +
  scale_alpha(guide="none") +
  theme_light() +
  theme(text = element_text(size = 16)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "bottom") +
  geom_dotplot(binaxis = 'y', size=4, fill = "black", color = "black", alpha = 1)


top / bottom 
```

<br>

<br>

Mutation rate estimates in coding vs. noncoding regions were not significantly different within the mysticetes (ANOVA p=`r round(myst_coding_aov[[1]][["Pr(>F)"]][1],digits = 2)`) or within the odontocetes (ANOVA p=`r round(odon_coding_aov[[1]][["Pr(>F)"]][1], digits = 2)`). While mutation rate estimates for these two regions do differ visually (Figure X), the variability between coding and non-coding regions is less within the mysticetes and odontocetes than the variability between the two suborders.


<br>

<br>

```{r coding_noncoding_mutation, fig.align = "center", fig.cap = "Figure 3. Mutation rate in coding vs. non-coding regions of the genome in both mysticetes and odontocetes."}


ggplot(data=code_noncode_data %>% filter(method == "noPI"), aes(x=suborder,y=rate,fill=snp_type, color=snp_type, alpha = 0.6)) + 
  geom_violin() +
  theme_light() +
  labs(y="Mutations/site/generation", x="") +
    scale_fill_manual(values=pnw_palette(n=8,name="Lake")[c(4,6)], 
                     name="SNP locus") +
  scale_color_manual(values=pnw_palette(n=8,name="Lake")[c(4,6)], 
                     name="SNP locus") +
  scale_alpha(guide="none") +
  theme(axis.title.y = element_text(margin = margin(t=0,r=8,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=8,r=0,b=5,l=0))) +
  theme(text = element_text(size = 16))
```

<br>

<br>

Similarly, mutation rates did not vary significantly among chromosomes within each species' genome (PVALUE HERE). While variability in mutation rate was qualitatively low among most chromosomes, especially compared to variability in mutation rate among species (Figure X), we observed that the mutation rate in Chromosome XX was lower across all species included in the study (Figure X). We also found that mutation rate variability among chromosomes was consistent across species - e.g. mutation rate was lower than the species-specific average in Chromosomes XX, XX, and XX in all species, and higher than the species-specific average in Chromosomes XX and XX in all species. 

Variability in mutation rate estimates by read depth here.

Mutation rate variation with lifespan here.

<br>

<br>

```{r whole_genome_models, fig.align = "center", fig.cap = "Figure X. Mutation rate of whole genomes and the lifespans of their samples alongside a linear model."}

ggplot(data=species_mrate_gen, aes(x=lifespan,y=rate)) +
  geom_point(aes(color=factor(infraorder)), size=3.7) +
  labs(x="Lifespan", y="Mutations/site/generation") +
  theme_light() +
  theme(text = element_text(size = 16)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0))) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)],
                     name = "Infraorder") +
  geom_smooth(method = "lm", color = "black", size = 0.75, alpha = 0.20)
```

<br>

<br>

We did not find that variability in divergence date estimates had a significant effect on mutation rate estimates (PVALUE HERE). Qualitatively, the difference in mutation rate estimates calculated using three divergence date estimates was trivial compared to the variability among species or among reference genomes (Supplementary Figure 1). The same statistical and qualitative pattern was true when calculating the difference in mutation rate caused by whether or not PI is used in the calculation (PVALUE HERE, Supplementary Figure 2).

## Discussion {-}

Mutation rate variability was higher than expected among species. While estimates of divergence date and the inclusion of PI had minimal effect on mutation rate estimates (Supplemental Figures S1 and S2), the choice of reference genome had the greatest effect on species-specific estimates of whole genome mutation rate. Similarly, mutation rates were appreciably - but not significantly - different in coding vs. non-coding regions and across chromosomes, but this variability was not as great as whole genome variability in mutation rates estimated among species or using different reference genomes.

The reference-driven differences in mutation rate observed in this study may be caused by several factors. For example, it is possible that the sperm whale evolutionary lineage has an overall lower mutation rate than other evolutionary lineages used as reference in this study. It may also be possible that mutation rate accelerates and slows over time, and that mutation rates were higher in deep evolutionary time, slowed in mid-evolutionary time, and are marginally higher in shallow or contemporary evolutionary time.

The degree of variability driven by choice of reference genome - up to an order of magnitude in some cetacean species - brings with it the question of how to choose the correct reference genome for alignment of new genomic data. Because mutation rate estimates represent an average over the evolutionary time period since divergence from the reference species it may be best to choose a reference genome that reflects the time period of the question. For example, population genomics or other studies concerned with recent evolutionary time might choose a reference genome from a recently-diverged species, and estimate mutation rate based on the reference species for use in downstream analyses, e.g. PSMC or BEAST phylogenies.

Compared to whole-genome mutation rate estimates among species, estimates of mutation rates among chromosomes varied less. Chromosome XX was the exception to this, likely driven at least in part by 

Species-specific mutation rates evolve over time, and are affected by demographic parameters like lifespan, generation time, and population size. Pedigree-based mutation rate estimates are a promising emerging tool that produce highly accurate estimates of mutation rate via counts of de novo mutations in parent-offspring trios. However, important to keep in mind that these highly accurate estimates of mutation rate will reflect contemporary population demographics and may differ significantly from historical mutation rates. **More here, REFS from Suarez-Menendes et al and Bergeron et al, Sung et al.).** 

## REFERENCES {-#REFERENCES}
<div id="refs"></div>

<br>

<br>

