---
title: "Mutation rate variability in cetacean genomes"
author: "Amy M. Van Cise^1\\*†^, Sophia Garrote^1^, Annabel Beichman^2^, Phillip A. Morin^3^"


output: 
    html_document:
      fig_caption: yes
      
bibliography: "`r rbbt::bbt_write_bib('bibliography.json', overwrite = TRUE)`"

knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_dir = "docs",
  output_file = "index.html") })
---

1. School of Aquatic and Fishery Sciences, University of Washington, Seattle, WA
3. Genome Sciences, University of Washington, Seattle, WA
2. Southwest Fisheries Science Center, National Marine Fisheries Service, National Oceanic and Atmospheric Administration, La Jolla, CA

<br>

^\*^Corresponding author email: avancise\@gmail.com

<br>

Running page head: Genomic mutation rate in cetaceans

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)

library(tidyverse)
library(ggsci)
library(PNWColors)
library(pander)
library(patchwork)
```

```{r data, cache = FALSE, warning = FALSE, message = FALSE, results = 'hide'}

load("Mutation-rate-coding-noncoding.Rdata")
load("whole_mutation_rate_df.Rdata")

```


<br>

## Abstract {-}

Genomic mutation is the fundamental process driving evolutionary processes that result in contemporary species, subspecies, and populations. Thus, genomic studies of divergence, speciation, and evolutionary demographics rely on accurate estimates of mutation rate. Historically, mutation rates have been estimated using a small portion of the genome and/or averaged among multiple species within a family or order, but growing evidence indicates potentially high variability in mutation rate both across the genome and among species. Here, we examine variability in mutation rate within cetaceans using high-quality genomes from the Vertebrate Genome Project. We estimate mutation rate across a range of parameters, including choice of input reference genome, coding vs. non-coding regions, chromosome-specific, species-specific, and using a variety of read depths, divergence date estimate, and methods to calculate mutation rate, in order to determine which factors have the greatest effect on variability in mutation rate. We find that mutation rate is most variable among species, and consequently most affected by the choice of reference genome used in the alignment pipeline. Other parameters had trivial effects on mutation rate estimates in comparison. These results indicate the importance of thoughtful selection of the reference genome used in future cetacean genomic studies, in order to ensure accurate downstream analyses of evolutionary processes.

## Introduction {-}

As the fundamental process underlying evolution, mutation drives processes such as divergence among species and evolutionary adaptation. The rate at which mutations accrue will ultimately determine the rate at which species diverge from each other or adapt to changes in their environment. 

Contemporary estimates of species divergence and adaptation that are central to evolutionary and population genetics rely on accurate estimates of mutation rate. Biases in this estimate may be reflected in many downstream analyses that shape evolutionary biology and conservation, e.g. the timing of splits in genetic phylogenies, patterns of selective divergence as they relate to neutral variation, or the accuracy of evolutionary demographic analyses. For example, a recent analysis of baleen whale mutation rates based wild pedigrees indicates that mutation rates in these species is higher than previously estimated based on phylogenetic divergence; correcting the mitochondrial mutation rate in estimates of preexploitation North Atlantic humpback whale abundance reduces those estimates by 86% [@suarez-menendez_etal23] and considerably shifts our understanding of the recovery of that population. Further, early studies of mutation rate suggest a lack of direct relationship between mtDNA genetic diversity and population size [@nabholz_etal08, REFS 18 from Nabholz]. 

Mutation rates have been shown to vary widely across species [e.g. @Nabholz2009a; @bergeron_etal23; REFS 14, 43, 39 from Nabholz] and over time (REFS). Three broad hypotheses have been proposed to explain potential drivers of mutation rate variability among species. (1) The longevity hypothesis posits that mutation rate decreases with increased lifespan, allowing individuals to live for longer periods of time without adverse effects of the accumulation of harmful mutations over their lifetimes. (2) The generation time hypothesis, which posits that mutation rate XX. (3) The metabolic rate hypothesis posits that mutation rate increases with increasing metabolic rate (often proxied by body mass) due to the oxidative effects of metabolism on the cell.

A growing body of evidence indicates that mutation rate varies across the genome @ellegren_etal03; @hodgkinson_eyre-walker11, and that site-specific variability is only partially context-dependent. 

Further, there is little research examining the potential effect of input parameters in the genome alignment pipeline on the accurate tally of genomic mutations and downstream estimation of mutation rate. 

The increasing accessibility of genomic sequencing technologies and high-quality reference genomes for marine mammals (cite VGP here) has precipitated an rise in the use of genomic techniques to address previously unanswered questions about the ecology and conservation of these species. Because these species are rare, remote, elusive, and protected, it is often difficult to obtain the sample size needed to accurately detect and describe ecological processes and patterns. In cases such as this, whole genome analyses increase sensitivity and power to describe these patterns using a smaller sample set, thus making it a potentiall powerful tool for studies of evolutionary ecology in marine mammals. These genomic studies in marine mammals, e.g. descriptions of evolutionary phylogenies or historical demographics of species, currently rely on estimates of mutation rate that are averaged across a family of species and often based on only a small portion of the genome. 

Here, we explore mutation rate variability within the cetaceans, and describe the major drivers affecting variability in estimates of mutation rate, with the aim of providing insight and context into the estimation of an accurate and appropriate mutation rate in cetacean genomics studies. We examine process driven variability by estimating taxon- and chromosome-specific mutation rates, and comparing species-specific mutation rates with key demographic traits considered to affect mutation rate. We also examine potential drivers of observation error by varying key parameters in the pipeline used to align genomes and estimate a mutation rate. These input parameters include the choice of reference genome used for alignment, minimum and maximum read depth threshold for genotyping a locus, inclusion of coding or non-coding regions in the mutation rate estimate, divergence date estimate used in the mutation rate calculation, and whether or not ancestral heterozygosity is included in the mutation rate calculation.

## Methods {-}

Short-read sample genomes and reference files (Tables 1 and 2) were retrieved from NCBI GenBank using SRAToolkit. Additional Hi-C format sample genomes from the European Nucleotide Archive were obtained using the scripts provided by the archive for download. The sample files, directly after being downloaded, do not have flags demarking forward and reverse reads, so a reformatting command must be run using the BBMap program. Reference genomes also need additional processing before scripts can be run, they require indexing by bwa or bwa-mem2 for the short-read and Hi-C formats respectively. Lastly, bed files, which are required for the step that masks repeated reads in the genome, must be generated for the reference genomes.

```{r sample_info_table}

sample_table_format <- sample_table_data %>% 
  rename("Species" = species, "Abbreviation" = abbreviation, "Accession" = "accession..", "Infra-order reference" = close.reference) %>% 
  mutate(Database = c(rep("NCBI", 6),"ENA",rep("NCBI", 4), rep("ERR",3))) %>% 
  relocate(Database, .before = Accession)

pander(sample_table_format, split.table = Inf, table.caption.prefix = "Table 1:", caption = "Species abbreviation codes, genome database, and accession number for each species included in the present study.")

```

<br>

<br>

The short-read alignment script uses samtools, bcftools, bedtools, and bwa commands to align the sample genome to the reference genome and extract variants. Additional filtering steps using Picard’s repeat masking function alongside bedtools removes potential duplicate reads from the variants.

Some of the sample genomes used in this study, however, were not in short-read format, but analyzed using Hi-C techniques, requiring a different script. This script still used samtools, bcftools, and bedtools, as only the steps aligning the reference genome needed to be altered. Instead of bwa, the bwa-mem2 module was instead used in alignment, which required re-indexing of any reference fna files, and the generation of an fai file using samtools faidx. Additionally, this more complicated alignment used external perl scripts to combine reads, as well as Picard for adding read groups (define significance of read groups and combining reads?). Picard is also used similarly to the short-read script in repeat-read masking with bedtools (significance of repeat masking?). The both alignment scripts culminate in a vcf file format, which can then be used in post-alignment processing. The scripts are used for every genome of their respective types and repeated for the multiple references a script may be aligned to.

Post-alignment processing entails using the vcf scripts produced to extract the information needed to calculate whole genome mutation rates, as well as other analyses, such as chromosome-level mutation rate. The R script we are using to calculate mutation rate is derived from the Robinson et al. (2022) paper on vaquitas, with modifications made to loop through all the samples used in this study, as only one genome was used in the Robinson paper. This script requires the total number of SNPs, the number of heterozygous variants, the number of homozygous variants, the number of alternate allele variants, the species’ generation time, and the species’ divergence times to calculate mutation rate. The generation and divergence times are obtained externally from (insert examples of sources here?). As previously mentioned, the number of specific types of variants can be obtained from the vcf file format using lines of code created with the awk command. 

(insert mutation rate proper equation?)

## Results {-}

Whole genome mutation rate variability is high among individual species (Figure 1). In addition to inter-specific variability in mutation rate, we found that phylogenetic inertia accounted for a large portion of this variability, i.e. mutation rates were significantly higher in mysticetes than in odontocetes (p = `r format(summary(infraorder_aov)[[1]][1,5], digits = 2)`, Figure X).

<br>

<br>

```{r whole_genome_rate, fig.height = 8, fig.width = 8, fig.cap = "Figure 1. Top: Whole genome mutation rate for each cetacean species included in the study. Bottom: Mean mutation rate for mysticete and odontocete clades. Mutation rates in both subplots were estaimted using Pmac as the reference genome and divergence estimates from McGowen et al. 2020. The dotted line represents an odontocete nuclear mutation rate estimate from Dornburg et al. 2012."}
top <- ggplot(data=whole_mrate_graph, aes(x=abbrev,y=rate)) +
  geom_point(size=3.7) +
  labs(x="Species", y="Mutations/\nsite/generation") +
  theme_light() +
  theme(text = element_text(size = 16), plot.subtitle = element_text(size = 14)) +
  guides(color=FALSE) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0))) +
  scale_shape_discrete(guide="none") +
  geom_hline(data=est2, mapping=aes(yintercept=yint),linetype="dotdash") +
  scale_linetype_discrete(guide="none")

bottom <- ggplot(data=whole_mrate_graph, aes(x=infraorder, y=rate, fill=infraorder)) +
  geom_violin() +
  labs(x="Infraorder", y="Mutations/\nsite/generation") +
  theme_light() +
  theme(text = element_text(size = 16)) +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none") +
  geom_dotplot(binaxis = 'y', size=4, fill = "black")
  
top / bottom
```

<br>

<br>

The reference genome used affected mutation rate variablity by an order of magnitude in some species (Figure X). Reference genomes for this analysis were chosen to reflect three periods of evolutionary divergence: deep, medium, and shallow. Mutation rate was highest in most species when compared to the deep divergence reference genome, indicating that mutation rate in cetaceans may have been higher in deep evolutionary time than current evolutionary time. Alignment to the middle divergence reference genome resulted in the lowest mutation rate estimates, and alignment to the shallow divergence reference genome resulted in mutation rate estimates that were intermediate between the deep and middle divergence reference genomes.

<br>

<br>

```{r multi_refs, fig.height = 8, fig.width = 8, fig.cap = "Figure 2. Top: Mutation rate using different reference genomes with various levels of evolutionary distance for each sample with multiple references. The dotted line represents an odontocete nuclear mutation rate estimate from Dornburg et al. 2012. Bottom: Mutation references for each of the references used, Egla is the close reference for mysticetes and Oorc is the close reference for odontocete. Mutation rates in both subplots were estaimted using divergence estimates from McGowen et al. 2020."}

top <- ggplot(data=mult_refs_graph, aes(x=abbrev,y=rate,shape=div_est,color=ref_dist)) +
  geom_point(size=3.7) +
  labs(x="Species", y="Mutations/\nsite/generation") +
  theme_light() +
  theme(text = element_text(size = 16), plot.subtitle = element_text(size = 14)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none") +
  scale_shape_discrete(guide="none") +
  scale_color_manual(values=pnw_palette(n=3,name="Sunset2"), 
                     name="Reference Distance") +
  geom_hline(data=est2, mapping=aes(yintercept=yint),linetype="dotdash")
  
  
bottom <- mult_refs_graph %>% 
  arrange(rate) %>%
  mutate(reference = factor(reference, levels=c("Egla", "Oorc", "Pmac", "Hamb"))) %>%
  mutate(ref_dist = factor(ref_dist, levels=c("Close", "Middle", "Far"))) %>%
  ggplot(aes(x=reference, y=rate, fill = ref_dist)) +
  geom_violin() +
  labs(x="Reference genome", y="Mutations/\nsite/generation") +
  scale_fill_manual(values=pnw_palette(n=3,name="Sunset2"), 
                     name="Reference Distance") +
  theme_light() +
  theme(text = element_text(size = 16)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "bottom") +
  geom_dotplot(binaxis = 'y', size=4, fill = "black")


top / bottom 
```

<br>

<br>

Mutation rate estimates in coding vs. noncoding regions were not significantly different within the mysticetes (ANOVA p=`r round(myst_coding_aov[[1]][["Pr(>F)"]][1],digits = 2)`) or within the odontocetes (ANOVA p=`r round(odon_coding_aov[[1]][["Pr(>F)"]][1], digits = 2)`). While mutation rate estimates for these two regions do differ visually (Figure X), the variability between coding and non-coding regions is less within the mysticetes and odontocetes than the variability between the two suborders.


<br>

<br>

```{r coding_noncoding_mutation, fig.align = "center", fig.cap = "Figure 3. Mutation rate in coding vs. non-coding regions of the genome in both mysticetes and odontocetes."}


ggplot(data=code_noncode_data %>% filter(method == "noPI"), aes(x=suborder,y=rate,fill=snp_type, color=snp_type, alpha = 0.6)) + 
  geom_violin() +
  theme_light() +
  labs(y="Mutations/site/generation", x="") +
    scale_fill_manual(values=pnw_palette(n=8,name="Lake")[c(4,6)], 
                     name="SNP locus") +
  scale_color_manual(values=pnw_palette(n=8,name="Lake")[c(4,6)], 
                     name="SNP locus") +
  scale_alpha(guide="none") +
  theme(axis.title.y = element_text(margin = margin(t=0,r=8,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=8,r=0,b=5,l=0))) +
  theme(text = element_text(size = 16))
```

<br>

<br>

Similarly, mutation rates did not vary significantly among chromosomes within each species' genome (PVALUE HERE). While variability in mutation rate was qualitatively low among most chromosomes, especially compared to variability in mutation rate among species (Figure X), we observed that the mutation rate in Chromosome XX was lower across all species included in the study (Figure X). We also found that mutation rate variability among chromosomes was consistent across species - e.g. mutation rate was lower than the species-specific average in Chromosomes XX, XX, and XX in all species, and higher than the species-specific average in Chromosomes XX and XX in all species. 

Variability in mutation rate estimates by read depth here.

Mutation rate variation with lifespan here.

<br>

<br>

```{r whole_genome_models, fig.cap = "Figure X. Mutation rate of whole genomes and the lifespans of their samples alongside a linear model."}

ggplot(data=whole_mrate_graph, aes(x=lifespan,y=rate)) +
  geom_point(aes(color=factor(infraorder)), size=3.7) +
  labs(x="Lifespan", y="Mutations/site/generation") +
  theme_light() +
  theme(text = element_text(size = 16)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0))) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)],
                     name = "Infraorder") +
  geom_smooth(method = "lm", color = "black", size = 0.75, alpha = 0.20)
```

<br>

<br>

We did not find that variability in divergence date estimates had a significant effect on mutation rate estimates (PVALUE HERE). Qualitatively, the difference in mutation rate estimates calculated using three divergence date estimates was trivial compared to the variability among species or among reference genomes (Supplementary Figure 1). The same statistical and qualitative pattern was true when calculating the difference in mutation rate caused by whether or not PI is used in the calculation (PVALUE HERE, Supplementary Figure 2).

## Discussion {-}

Mutation rate variability was higher than expected among species. While estimates of divergence date and the inclusion of PI had minimal effect on mutation rate estimates (Supplemental Figures S1 and S2), the choice of reference genome had the greatest effect on species-specific estimates of whole genome mutation rate. Similarly, mutation rates were appreciably - but not significantly - different in coding vs. non-coding regions and across chromosomes, but this variability was not as great as whole genome variability in mutation rates estimated among species or using different reference genomes.

The reference-driven differences in mutation rate observed in this study may be caused by several factors. For example, it is possible that the sperm whale evolutionary lineage has an overall lower mutation rate than other evolutionary lineages used as reference in this study. It may also be possible that mutation rate accelerates and slows over time, and that mutation rates were higher in deep evolutionary time, slowed in mid-evolutionary time, and are marginally higher in shallow or contemporary evolutionary time.

The degree of variability driven by choice of reference genome - up to an order of magnitude in some cetacean species - brings with it the question of how to choose the correct reference genome for alignment of new genomic data. Because mutation rate estimates represent an average over the evolutionary time period since divergence from the reference species it may be best to choose a reference genome that reflects the time period of the question. For example, population genomics or other studies concerned with recent evolutionary time might choose a reference genome from a recently-diverged species, and estimate mutation rate based on the reference species for use in downstream analyses, e.g. PSMC or BEAST phylogenies.

Compared to whole-genome mutation rate estimates among species, estimates of mutation rates among chromosomes varied less. Chromosome XX was the exception to this, likely driven at least in part by 

Species-specific mutation rates evolve over time, and are affected by demographic parameters like lifespan, generation time, and population size. Pedigree-based mutation rate estimates are a promising emerging tool that produce highly accurate estimates of mutation rate via counts of de novo mutations in parent-offspring trios. However, important to keep in mind that these highly accurate estimates of mutation rate will reflect contemporary population demographics and may differ significantly from historical mutation rates. **More here, REFS from Suarez-Menendes et al and Bergeron et al, Sung et al.).** 

## REFERENCES {-#REFERENCES}
<div id="refs"></div>

<br>

<br>

