---
title: "Mutation rate variability in cetacean genomes"
author: "Amy M. Van Cise^1\\*â€ ^, Sophia Garrote^1^, Annabel Beichman^2^, Phillip A. Morin^3^"


output: 
    html_document:
      fig_caption: yes
      
bibliography: "`r rbbt::bbt_write_bib('bibliography.json', overwrite = TRUE)`"

knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_dir = "docs",
  output_file = "index.html") })
---

1. School of Aquatic and Fishery Sciences, University of Washington, Seattle, WA
3. Genome Sciences, University of Washington, Seattle, WA
2. Southwest Fisheries Science Center, National Marine Fisheries Service, National Oceanic and Atmospheric Administration, La Jolla, CA

<br>

^\*^Corresponding author email: avancise\@gmail.com

<br>

Running page head: Genomic mutation rate in cetaceans

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)

library(tidyverse)
library(ggsci)
library(PNWColors)
library(pander)
library(patchwork)
library(ggbeeswarm)
```

```{r data, cache = FALSE, warning = FALSE, message = FALSE, results = 'hide'}

load("Mutation-rate-coding-noncoding.Rdata")
load("whole_mutation_rate_df.Rdata")
load("mrate_models_df.Rdata")

```


<br>

## Abstract {-}

Genomic mutation is the fundamental process driving evolutionary processes that result in contemporary species, subspecies, and populations. Thus, genomic studies of divergence, speciation, and evolutionary demographics rely on accurate estimates of mutation rate. Historically, mutation rates have been estimated using a small portion of the genome and/or averaged among multiple species within a family or order, but growing evidence indicates potentially high variability in mutation rate both across the genome and among species. Here, we examine variability in mutation rate within cetaceans using high-quality genomes from the Vertebrate Genome Project. We estimate mutation rate across a range of parameters, including choice of input reference genome, coding vs. non-coding regions, chromosome-specific, species-specific, and using a variety of read depths, divergence date estimate, and methods to calculate mutation rate, in order to determine which factors have the greatest effect on variability in mutation rate. We find that mutation rate is most variable among species, and consequently most affected by the choice of reference genome used in the alignment pipeline. Other parameters had trivial effects on mutation rate estimates in comparison. These results indicate the importance of thoughtful selection of the reference genome used in future cetacean genomic studies, in order to ensure accurate downstream analyses of evolutionary processes.

## Introduction {-}

As the fundamental process underlying evolution, mutation drives processes such as divergence among species and evolutionary adaptation. The rate at which mutations accrue will ultimately determine the rate at which species diverge from each other or adapt to changes in their environment. 

Contemporary estimates of species divergence and adaptation that are central to evolutionary and population genetics rely on accurate estimates of mutation rate. Biases in this estimate may be reflected in many downstream analyses that shape our understanding of evolutionary biology, e.g. the timing of splits in genetic phylogenies, patterns of selective divergence as they relate to neutral variation, or the accuracy of evolutionary demographic analyses. They may also have affect conservation priorities and management actions. For example, a recent analysis of baleen whale mutation rates based wild pedigrees indicates that mutation rates in these species is higher than previously estimated based on phylogenetic divergence; correcting the mitochondrial mutation rate previously used in genetic estimates of pre-exploitation North Atlantic humpback whale abundance reduces those estimates by 86% [@suarez-menendez_etal23] and considerably shifts our understanding of the recovery of that population. 

Evolutionary demographic analysis relate genetic diversity, or the number of mutations in a given genetic lineage, to abundance via the coalescent theory. In these types of analyses especially, the mutation rate estimate used may have a drastic impact on our understanding of historic abundance of a population - and changes in the mutation rate over time could also affect the observed trends in abundance over time. Early studies of mutation rate suggest a lack of direct relationship between mtDNA genetic diversity and population size [@nabholz_etal08, REFS 18 from Nabholz]. Further, a growing body of evidence indicates that mutation rate varies across the genome, and that site-specific variability is only partially context-dependent [@ellegren_etal03; @hodgkinson_eyre-walker11]. Finally, mutation rate has been shown to vary over evolutionary timescales (REF).

In addition to varying throughout the genome and through time, mutation rates have been shown to vary widely across species [e.g. @Nabholz2009a; @bergeron_etal23; REFS 14, 43, 39 from Nabholz]. Three broad hypotheses have been proposed to explain potential drivers of mutation rate variability among species. (1) The longevity hypothesis posits that mutation rate decreases with increased lifespan, allowing individuals to live for longer periods of time without adverse effects of the accumulation of harmful mutations over their lifetimes. (2) The generation time hypothesis, which posits that mutation rate XX. (3) The metabolic rate hypothesis posits that mutation rate increases with increasing metabolic rate (often proxied by body mass) due to the oxidative effects of metabolism on the cell.

The effects of demography, phylogeny, and genome location on mutation rate highlight the need to quantify mutation rate variability within taxonomic groups in order to inform downstream evolutionary and conservation genetic analyses. In addition to these factors, there are multiple decision points along the bioinformatic pipeline from sequence data to aligned genome that may affect the accurate tally of genomic mutations and downstream estimation of mutation rate. However, there is limited research on how the selection of these input parameters affects mutation rate estimates.

The increasing accessibility of genomic sequencing technologies and high-quality reference genomes for marine mammals (cite VGP here) has precipitated an rise in the use of genomic techniques to address previously unanswered questions about the ecology and conservation of these species. Because these species are rare, remote, elusive, and protected, it is often difficult to obtain the sample size needed to accurately detect and describe ecological processes and patterns. In cases such as this, whole genome analyses increase sensitivity and power to describe genetic patterns using a smaller sample set, thus making it a potentially powerful tool for studies of evolutionary ecology in marine mammals. These genomic studies in marine mammals, e.g. descriptions of evolutionary phylogenies or historical demographics of species, currently rely on estimates of mutation rate that are averaged across a family of species and often based on only a small portion of the genome. 

Here, we explore mutation rate variability within the cetaceans, and describe the major drivers affecting variability in estimates of mutation rate, with the aim of providing insight and context into the estimation of an accurate and appropriate mutation rate in cetacean genomics studies. We examine process driven variability by estimating taxon- and chromosome-specific mutation rates, and comparing species-specific mutation rates with key demographic traits considered to affect mutation rate. We also examine potential drivers of observation error by varying key parameters in the pipeline used to align genomes and estimate a mutation rate. These input parameters include the choice of reference genome used for alignment, minimum and maximum read depth threshold for genotyping a locus, inclusion of coding or non-coding regions in the mutation rate estimate, divergence date estimate used in the mutation rate calculation, and whether or not ancestral heterozygosity is included in the mutation rate calculation.

## Methods {-}

### Sample and reference genome preparation {-}

We acquired genomes from 14 cetacean species, representing the breadth of cetacean phylogenetic diversity, from NCBI Genbank and the European Nucleotide Archive using SRAToolkit (REF, Table 1). Genomic data were generated by the Vertebrate Genome Progect (REF) using either Illumina short-read or Dovetail Hi-C approaches (REFS?). Prior to sequence assembly, we formatted all sequence files using BBMap (REF) to flag forward and reverse reads. Sequence files were formatted  assembly using BBMap (REF). 

In order to test for the effect of reference genome on the resulting mutation rate, we selected four reference genomes representing a deeply diverged taxon, a recently diverged taxon, and an intermediate taxon (Table 2). The deeply diverged taxon is represented by the hippopotamus (*Hippopotamus amphibius*), and the intermediate taxon is represented by the sperm whale (*Physeter macrocephaluS*). We selected two species to represent the recently diverged taxon: the killer whale (*Orcinus orca*) is used for all odontocetes, and the North Atlantic right whale (*Eubalaena glacialis*) is used for all mysticetes.

We downloaded all reference genomes as fastq files from NCBI Genbank, and created index files using bwa (REF). Sample genomes that were generated using Dovetail Hi-C require additional index files, which we generated using bwa-mem2 and samtools (REFS). Finally, we mapped repeats and created a repeat masker for each reference file using bedtools (REF). All steps described here, making up the pre-alignment pipeline, are outlined at: https://github.com/UW-WADE-lab/Whole-Genome-Alignment

```{r sample_info_table}

sample_table_format <- sample_table_data %>% 
  rename("Species" = species, "Abbreviation" = abbreviation, "Accession" = "accession..", "Recent reference" = close.reference) %>% 
  mutate(Database = c(rep("NCBI", 6),"ENA",rep("NCBI", 4), rep("ENA",3))) %>% 
  relocate(Database, .before = Accession)

pander(sample_table_format, split.table = Inf, table.caption.prefix = "Table 1:", caption = "Species abbreviation codes, genome source database, and accession number for each species included in the present study. For all species included in the multiple reference comparison, the final column indicates which species was useds as the recently diverged reference. **Add columns with generation time and reference for each species**")

pander(reference_table_data, split.table = Inf, table.caption.prefix = "Table 2:", caption = "Species abbreviation codes, genome source database, and accession number for each species used as a reference genome for alignment in this study.")

```

<br>

<br>

### Sample alignment {-}

All 14 sample genomes were first aligned to the intermediate reference genome, the sperm whale. The alignment pipeline, regardless of sequencing approach, comprised six steps: 1) align sample genome to reference, 2) sort aligned genome, 3) remove duplicate sequences, 4) mask repeat regions, 5) index genome, 6) call genotypes and filter variants by read depth. All alignment scripts can be found at https://github.com/UW-WADE-lab/Whole-Genome-Alignment.

Step 1 differed for Illumina short-read and Dovetail Hi-C sample genomes. We aligned forward and reverse reads from Illumina short-read genomes using bwa (REF). Dovetail Hi-C genomes, which comprise longer reads that are vulnerable to chimeras resulting from hairpin turns (?) (REF), require an additional filtering step post-alignment. Therefore, we aligned Dovetail Hi-C forward and reverse reads separately using bwa-mem2, and filtered forward and reverse read files separately using a custom perl script generated by Arima (REF/website here?) to identify and remove chimeras. We then combined forward and reverse reads using a custom perl script from Arima (REF/website here?).

Steps 2-6 were identical for both types of sample genome. We sorted genomes using samtools (REF) and removed duplicates using Picard (REF). We masked repeat regions, previously identified using the reference genome, with bedtools (REF). We indexed the sample genome using samtools, called genotypes using bcftools (REF), and filtered variants to remove those with a read depth greater than 500, also using bcftools. Note that this read depth filter is higher than is normally used in an alignment pipeline; this allowed us to directly examine the effect of read depth on mutation rate later in the study. 

We selected seven of the 14 sample genomes for a comparison of mutation rate estimates among reference genomes. These samples were additionally aligned to the hippopotamus and either the killer whale (odontocetes) or the North Atlantic right whale (mysticetes) using the pipeline described above (Table 1).

### Data analysis: calculating mutation rate {-}

Mutation rate ($\mu$) is commonly estimated as genomic divergence (i.e. genomic heterozygosity, $d_{xy}$) over divergence time ($t$):

\begin{equation}
    \mu=d_{xy}/2t
\end{equation}

Genomic divergence ($d_{xy}$), in turn, is estimated as:

\begin{equation}
    d_{xy}=(h_1+2h_2)/2g
\end{equation}

where $h_1$ is the number of sites in which the sample genome is heterozygous, $h_2$ is the number of sites in which the sample genome is homozygous for the alternate allele, and $g$ is the number of sites in which both the sample and reference genomes had called genotypes.

This calculation may result in an overestimate of mutation rate because ancestral (pre-divergence) heterozygosity is not taken into account, which we can account for in the following:

\begin{equation}
    \mu=(d_{xy}-\pi_{anc})/2t
\end{equation}

The above equations produce a per-year mutation rate estimate. Mutation rates are often reported per generation in the literature, so to consider the effect of including generation time in mutation rate estimates we converted our per-year mutation rates to an estimate of mutation rate per generation. For most species, generation time was available from Taylor et al. [-@Taylor2007, Table 1].

### Data analysis

All data analyses were completed in R [@RCoreTeam2016] using Rstudio (REF). We subset our mutation rate estimates according to each test described below. Unless otherwise noted, for consistency among analyses we ensured that analysis datasets included genomes from all 14 species considered in the study, and we subset our mutation rate dataset to include estimates only from genomes aligned to the sperm whale reference genome, using divergence date estimates from McGowen et al. [-@mcgowen_etal20] and without correcting for ancestral heterozygosity. Using a Shapiro-Wilk normality test [@royston82; @royston82a], we determined that this subsetted mutation rate dataset is normally distributed (p = `r round(rate_normality$p.value, digits = 2)`), therefore we used ANOVA [@hollander_etal15p] for all tests of mutation rate differentiation among categorical taxonomic strata, and a Guassian distribution for our regression analyses of demographic effects.

### Data analysis: pipeline effects on mutation rate {-}

We first consider the effect of the following sources of potential variability in the genome alignment pipeline on downstream estimates of mutation rate:

1. Choice of reference species for genomic alignment
2. Genomic location of regions used in estimate (i.e. chromosome-level mutation rate variability)
3. Inclusion of coding vs. non-coding regions in estimate
4. Maximum read depth cutoff for called genotypes

We also consider the effect of variability in two estimated parameters that are used in the calculation of mutation rate: 

1. Choice of divergence date estimates
2. Choice of whether and how to correct for ancestral heterozygosity

**Reference species:** As described above, we additionally aligned seven of the fourteen species to a deeply diverged reference (hippopotamus) and a recently diverged reference (North Atlantic right whale or killer whale, see Table 2). We when grouped whole genome mutation rates from these seven species into three categories: deep divergence, intermediate divergence, and recent divergence, and used an ANOVA analysis to determine whether mutation rate varies significantly among these three groups.

**Chromosome position:** We extracted SNPs located on each chromosome by mapping to a bed file generated from annotations of the sperm whale reference genome (available from NCBI Genbank) and estimated a chromosome-specific mutation rate for each species. We then used an ANOVA analysis to determine whether mutation rate varies significantly among chromosomes across all species included in the study.

**Coding vs non-coding regions:** We extracted SNPs located in coding and non-coding regions by mapping to a bed file generated from annotations of the sperm whale reference genome (available from NCBI Genbank) to estimate mutation rates in coding and non-coding regions. We then compared mutation rate estimates between these two regions, first across all fourteen species included in the study, and then within the mysticetes and the odontocetes.

**Genotype read-depth:** We first generated new genotype vcf files with calls for all loci, including both heterozygous and homozygous loci. Using these vcf files and considering read depths of 10-500 reads, we then used a custom R script to determine the cumulative proportion of heterozygous sites with increasing read depth. Following this, we calculated mutation rate for each species at maximum read depth cutoffs from 100-500. Finally, we use a generalized linear regression to test whether read depth is a significant driver of mutation rate estimates. Because phylogenetic inertia may be a significant factor affecting variability in mutation rate among species, we used a phylogenetic generalized least squares (PGLS) modeling approach implemented using the pgls package in R (REF), and chose the best model and significant parameters based on reported AIC and p values, respectively.

**Divergence date estimates:** We compare estimates of mutation rate using fossil-dated divergence times ($t$) from McGowen et al. [-@mcgowen_etal20] and Lloyd and Slater [-@lloyd_slater21], as well as an average across all known divergence times from TimeTree [timetree.org; @kumar_etal17].

**Ancestral heterozygosity:** Because the ancestral heterozygosity ($\pi_{anc}$) of a species is unknown and unknowable using currently available technology, we compare mutation rate estimates under five different states of $\pi_{anc}$. First, we estimate mutation rate ($\mu$) without correcting for $\pi_{anc}$, and by correcting for $\pi_{anc}$ under the assumption that the contemporary genomic heterozygosity of each species is equivalent to its ancestral heterozygosity, i.e., using a whole-genome estimate of heterozygosity from each species to correct for $\pi_{anc}$. We further test the sensitivity of our estimates of $\mu$ to a range of values of $\pi_{anc}$ by re-estimating $\mu$ for every species using the maximum and minimum reported values of $\pi_{anc}$ within cetaceans from Morin et al. (REF here).

### Data analysis: phylogenetic or demographic effects on mutation rate {-}

To examine the effect of phylogenetic inertia on mutation rate, we test for significant differentiation among taxonomic families in whole-genome mutation rate per year (mutations/site/year) as well as whole-genome mutation rate per generation (mutations/site/generation). We also test for rate differentiation among three families for which we have more than one species: Balaenopteridae, Delphinidae, and Ziphiidae. 

Finally, we also consider potential effects of lifespan and body size on whole-genome mutation rate using data on species-specific maximum lifespan from XXXX (REF) and maximum body length from XXX (REF). We first tested for correlation between lifespan, maximum body length, and generation time and found them to be correlated in our dataset (VALUE HERE). Next, we fit generalized linear regressions considering all possible combinations of these three demographic parameters using the PGLS modeling approach described above.

For our examination of both phylogenetic and demographic effects on mutation rate, we additionally compare the patterns observed using both annual and generational estimates of mutation rate, which are both commonly reported in marine mammal genomic literature (REFS here).

## Results {-}

For each combination of species and reference genome, we generated 18 estimates of whole-genome mutation rate by varying estimates of divergence date, ancestral heterozygosity, and annual vs. generational mutation rate estimates. Using the sperm whale reference genome, divergence date from McGowen et al. [-@mcgowen_etal20], and without correcting for ancestral heterozygosity, we also generated an estimate of mutation rate at each chromosome (n = 20 per species), in coding and non-coding regions (n = 2 per species), and at maximum read depths of 100-500 (n = XX per species).

Whole genome mutation rate varies among individual species included in the study, and is overall lower than previously published estimates for all odontocetes or XXXX (Figure 1).

<br>

<br>

```{r whole_genome_rate, fig.align = "center", fig.height = 8, fig.width = 8, fig.cap = "Figure 1. Top: Whole genome mutation rate for each cetacean species included in the study. Bottom: Mean mutation rate for mysticete and odontocete clades. Mutation rates in both subplots were estimated using Pmac as the reference genome and divergence estimates from McGowen et al. 2020. The dotted line represents an odontocete nuclear mutation rate estimate from Dornburg et al. 2012."}
ggplot(data=species_mrate_year, aes(x=abbrev,y=rate)) +
  geom_point(size=3.7) +
  labs(x="Species", y="Mutations/\nsite/year") +
  theme_light() +
  theme(text = element_text(size = 16), plot.subtitle = element_text(size = 14)) +
  guides(color=FALSE) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0))) +
  scale_shape_discrete(guide="none") +
  geom_hline(aes(yintercept=2.2e-10), linetype="dotdash") +
  scale_linetype_discrete(guide="none")
```

### Pipeline effects on mutation rate

#### Reference species
The reference genome used had the greatest overall magnitude of effect among all factors tested in this study (Figure 2, top). Reference genomes for this analysis were chosen to reflect three periods of evolutionary divergence: deep divergence, intermediate divergence, and recent divergence. Mutation rate per year was highest in most species when compared to the deep divergence reference genome, indicating that mutation rate in cetaceans may have been higher in deep evolutionary time than current evolutionary time. Alignment to the intermediate divergence reference genome resulted in the lowest mutation rate estimates across all species in the study, and alignment to the recent divergence reference genome resulted in mutation rate estimates that were between the deep and intermediate reference genomes. 

Yearly mutation rate estimates did not differ significantly across mysticetes and odontocetes when aligned to the deep reference genomes, but did vary significantly when aligned to the intermediate and recently diverged genomes, with higher annual mutation rates in odontocetes than in mysticetes (Figure 2, bottom).

<br>

<br>

```{r multi_refs,  fig.align = "center", fig.height = 8, fig.width = 8, fig.cap = "Figure 2. Top: Mutation rate using different reference genomes with various levels of evolutionary distance for each sample with multiple references. The dotted line represents an odontocete nuclear mutation rate estimate from Dornburg et al. 2012. Bottom: Mutation references for each of the references used, Egla is the close reference for mysticetes and Oorc is the close reference for odontocete. Mutation rates in both subplots were estaimted using divergence estimates from McGowen et al. 2020."}

top <- ggplot(data=mult_refs_year, aes(x=abbrev,y=rate,color=ref_dist)) +
  geom_point(size=3.7) +
  labs(x="Species", y="Mutations/site") +
  theme_light() +
  #facet_wrap(~time_step, scales = "free_y") +
  theme(text = element_text(size = 16)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "bottom") +
  scale_color_manual(values=pnw_palette(n=3,name="Sunset2"), 
                     name="Reference Distance") 
  
  
bottom <- mult_refs_year %>% 
  arrange(rate) %>%
  mutate(reference = factor(reference, levels=c("Egla", "Oorc", "Pmac", "Hamb"))) %>%
  mutate(ref_dist = factor(ref_dist, levels=c("Recent", "Intermediate", "Deep"))) %>%
  ggplot(aes(x=ref_dist, y=rate, fill = infraorder, color = infraorder, alpha = 0.6)) +
  geom_violin(position = position_dodge(width = 0.5)) +
  #facet_wrap(~time_step, scales = "free_y") +
  labs(x="Reference genome", y="Mutations/site") +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)],
                     name = "Infraorder") +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)],
                     name = "Infraorder") +
  # scale_fill_manual(values=pnw_palette(n=3,name="Sunset2"), 
  #                    name="Reference Distance") +
  # scale_color_manual(values=pnw_palette(n=3,name="Sunset2"), 
  #                    name="Reference Distance") +
  scale_alpha(guide="none") +
  theme_light() +
  theme(text = element_text(size = 16)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "bottom") +
  #geom_dotplot(binaxis = 'y', dotsize=0.75, fill = "black", color = "black",shape = species, alpha = 0.5, stackdir = "center", position = "dodge")
  #geom_point(position = position_jitterdodge(seed = 1, dodge.width = 0.5), fill = "black", color = "black", alpha = 0.5, shape = species)
  geom_quasirandom(dodge.width = 0.5, varwidth = TRUE, color = "black")

top / bottom 
```

<br>

<br>

#### Chromosome position

<br>

<br>

Mutation rates did not vary significantly among chromosomes within each species' genome (PVALUE HERE). While variability in mutation rate was qualitatively low among most chromosomes, especially compared to variability in mutation rate among species (Figure X), we observed that the mutation rate in Chromosome XX was lower across all species included in the study (Figure X). We also found that mutation rate variability among chromosomes was consistent across species - e.g. mutation rate was lower than the species-specific average in Chromosomes XX, XX, and XX in all species, and higher than the species-specific average in Chromosomes XX and XX in all species. 

<br>

<br>

```{r chromosome_position, fig.align = "center", fig.cap = "Figure 3. Mutation rate across chromosome of the genomes of each species when aligned to the sperm whale reference genome."}

```

<br>

<br>

#### Coding vs. non-coding regions
Annual mutation rate estimates were significantly different in coding vs. noncoding regions (ANOVA p=`r round(summary(code_noncode_aov)[[1]][["Pr(>F)"]][1],digits = 8)`, F = `r round(summary(code_noncode_aov)[[1]][["F value"]][1],digits = 2)`,  Figure 4). Rates were higher in coding regions than non-coding regions across all species included in the study, increasing by an average of `r round(mean(coding_difference$diff_rate), digits = 12)` mutations/site/year (min = `r round(min(coding_difference$diff_rate), digits = 12)`, max = `r round(max(coding_difference$diff_rate), digits = 12)`).

<br>

<br>

```{r coding_noncoding_mutation, fig.align = "center", fig.cap = "Figure 4. Annual mutation rate (mutations/site/year) in coding vs. non-coding regions of the genome."}

ggplot(data=code_noncode_data %>% filter(method == "noPI"), aes(x=snp_type,y=rate,fill=snp_type, color=snp_type, alpha = 0.6)) + 
  geom_violin() +
  theme_light() +
  labs(y="Mutations/site/year", x="Genome region") +
  scale_fill_manual(values=pnw_palette(n=8,name="Lake")[c(4,6)], 
                     name="SNP locus") +
  scale_color_manual(values=pnw_palette(n=8,name="Lake")[c(4,6)], 
                      name="SNP locus") +
  scale_alpha(guide="none") +
  theme(axis.title.y = element_text(margin = margin(t=0,r=8,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=8,r=0,b=5,l=0))) +
  theme(text = element_text(size = 16)) +
  geom_quasirandom(dodge.width = 0.000001, color = "black")
  
```

#### Genotyping read depth

#### Divergence date estimates

Varying divergence date estimates also had a significant effect on annual mutation rate estimates, although the magnitude of the effect is smaller than the effect of reference genome or coding vs. non-coding regions (ANOVA p=`r round(summary(div_anova)[[1]][["Pr(>F)"]][1],digits = 2)`, F = `r round(summary(div_anova)[[1]][["F value"]][1],digits = 2)`,  Figure 6).

<br>

<br>

```{r divergence, fig.align = "center", fig.cap = "Figure 6. Annual mutation rate (mutations/site/year) estimated using divergence times from McGowen et al. 2020, Lloyd and Slater XXXX, and Timetree.org."}

ggplot(data = div_test_data, aes(x=div_est, y=rate, fill = div_est, color = div_est, alpha = 0.6)) +
  geom_violin() +
  theme_light() +
  xlab("Divergence date estimate") +
  ylab("Mutations/site/year") +
  theme(text = element_text(size = 16)) +
  scale_fill_manual(values=pnw_palette(n=5, name="Winter")) +
  scale_color_manual(values=pnw_palette(n=5, name="Winter")) +
  scale_alpha(guide="none") +
  geom_quasirandom(dodge.width = 1, varwidth=TRUE, color = "black") +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none")

```

<br>

<br>

#### Ancestral heterozygosity
Varying divergence date estimates also had a significant effect on annual mutation rate estimates; similar to the effect of divergence date, the magnitude of the effect of ancestral heterozygosity is smaller compared to effect of reference genome or coding vs. non-coding regions (ANOVA p=`r round(summary(pi_anova)[[1]][["Pr(>F)"]][1],digits = 2)`, F = `r round(summary(pi_anova)[[1]][["F value"]][1],digits = 2)`,  Figure 7).

<br>

<br>

```{r anc_het, fig.align="center", fig.cap = "Figure 7. Annual mutation rate (mutations/site/year) estimated using five approaches to accounting for ancestral heterozygosity: 1) no $\\pi_{anc}$, 2) $\\pi_{anc}$ equal to contemporary heterozygosity, 3) $\\pi_{anc}$ equal to mean cetacean heterozygosity, 4) $\\pi_{anc}$ equal to minimum cetacean heterozygosity, and 5) $\\pi_{anc}$ equal to maximum cetacean heterozygosity."}

ggplot(data = pi_test_data, aes(x=Pimethod, y=rate, fill = Pimethod, color = Pimethod, alpha = 0.6)) +
  geom_violin() +
  theme_light() +
  theme(text = element_text(size = 16)) +
  scale_fill_manual(values=pnw_palette(n=5, name="Shuksan")) +
  scale_color_manual(values=pnw_palette(n=5, name="Shuksan")) +
  scale_alpha(guide="none") +
  geom_quasirandom(dodge.width = 0.000001, color = "black") +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none")
  
```

<br>

<br>

#### Comparison among all pipeline effects

Qualitatively comparing F-ratios among all ANOVA tests (Table 2) indicates that the pipeline choices having the greatest effect on mutation rate estimates are reference species/reference distance and whether/how to correct for ancestral heterozygosity, followed by the choice of whether to including coding regions, non-coding regions, or both.

```{r comparison, fig.align = "center", fig.cap = "Table 2. Comparison of ANOVA F-ratios among all pipeline effects tested in this study."}

effect_data <- data.frame(effect = c("reference distance", "reference species", "chromosome position", "coding/noncoding", "read depth", "divergence date", "ancestral heterozygosity"), size = c(summary(reference_anova)[[1]][["F value"]][1], summary(reference_anova_sp)[[1]][["F value"]][1], "unknown", summary(code_noncode_aov)[[1]][["F value"]][1], "unknown", summary(div_anova)[[1]][["F value"]][1], summary(pi_anova)[[1]][["F value"]][1]))

pander(effect_data, round = c(NA,2))
```

### Phylogenetic or demographic effects on mutation rate

We first examined the effect of phylogenetic inertia on mutation rate, and found that annual mutation rate was not significantly different between the two cetacean infraorder when including the Amazon river dolphin, which has an anomalously low mutation rate (see Figure 1), within the odontocete infraorder (ANOVA p = `r format(summary(infraorder_anova_year)[[1]][1,5], digits = 2)`, F = `r round(summary(infraorder_anova_year)[[1]][["F value"]][1],digits = 2)`, Figure 8a). However, when excluding the Amazon river dolphin from the odontocete infraorder we find that annual mutation rate is significantly different between infraoders (ANOVA p = `r format(summary(infraorder_anova_year_sub)[[1]][1,5], digits = 2)`, F = `r round(summary(infraorder_anova_year_sub)[[1]][["F value"]][1],digits = 2)`, Figure 8a inset) and accounts for approximately `r round(summary(infraorder_anova_year_sub)[[1]][1,2]/(summary(infraorder_anova_year_sub)[[1]][1,2]+summary(infraorder_anova_year_sub)[[1]][2,2]), digits = 2)*100`% of annual mutation rate variability, with mutation rates that were significantly lower in mysticetes than in odontocetes. We also found significant differences in annual mutation rate among the three families with 2 or more species included in this study: Balaenopteridae, Delphinidae, and Ziphiidae (ANOVA p = `r round(summary(family_anova)[[1]][1,5], digits = 6)`, F = `r round(summary(family_anova)[[1]][["F value"]][1],digits = 2)`).

We then re-analyzed mutation rate between infraorders and among families using generational mutation rates, and found no significant differentiation in either stratification (infraorder ANOVA p = `r format(summary(infraorder_anova_gen)[[1]][1,5], digits = 2)`, subset infraorder ANOVA p = `r format(summary(infraorder_anova_gen_sub)[[1]][1,5], digits = 2)`, family ANOVA p = `r round(summary(family_anova_gen)[[1]][1,5], digits = 2)`, Figure 8c+d). Further, the observed pattern in mutation rate differentiation between infraorders is flipped, with a higher mean generational mutation rate in the mysticetes than in the odontocetes, both with and without the Amazon river dolphin (Figure 8c).

<br>

<br>

```{r phylogenic_effects, fig.align = "center", fig.cap = "Figure 8. Left: Variability in annual mutation rate (mutations/site/year) estimated in two infraorders (top) and three families (bottom). Right: Variability in generational mutation rate (mutations/site/generation) estimated in two infraorders (top) and three families (bottom)."}

fig8a <- ggplot(data=species_mrate_year, aes(x=infraorder, y=rate, fill=infraorder, color=infraorder, alpha=0.6)) +
  geom_violin() +
  labs(x=NULL, y=NULL) +
  theme_light() +
  theme(text = element_text(size = 14)) +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_alpha(guide="none") +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none") +
  geom_quasirandom(dodge.width = 0.000001, color = "black")

fig8ainset <- ggplot(data=species_mrate_year %>% filter(abbrev != "Igeo"), aes(x=infraorder, y=rate, fill=infraorder, color=infraorder, alpha=0.6)) +
  geom_violin() +
  theme_light() +
  theme(text = element_text(size = 14), legend.position = "none",
        axis.title = element_blank(),
        axis.text = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "null"),
        panel.margin = unit(c(0, 0, 0, 0), "null"),
        axis.ticks = element_blank()) +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_alpha(guide="none") +
  geom_quasirandom(dodge.width = 0.000001, color = "black")

fig8b <- ggplot(data=family_data, aes(x=family, y=rate, fill=family, color=family, alpha=0.6)) +
  geom_boxplot() +
  labs(x=NULL, y=NULL) +
  theme_light() +
  theme(text = element_text(size = 14)) +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(1,4,5)]) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(1,4,5)]) +
  scale_alpha(guide="none") +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none") +
  geom_quasirandom(dodge.width = 1, color = "black") 

fig8c <- ggplot(data=species_mrate_gen, aes(x=infraorder, y=rate, fill=infraorder, color=infraorder, alpha=0.6)) +
  geom_violin() +
  labs(x=NULL, y=NULL) +
  theme_light() +
  theme(text = element_text(size = 14)) +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_alpha(guide="none") +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none") +
  geom_quasirandom(dodge.width = 0.000001, color = "black")

fig8cinset <- ggplot(data=species_mrate_gen %>% filter(abbrev != "Igeo"), aes(x=infraorder, y=rate, fill=infraorder, color=infraorder, alpha=0.6)) +
  geom_violin() +
  theme_light() +
  theme(text = element_text(size = 14), legend.position = "none",
        axis.title = element_blank(),
        axis.text = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "null"),
        panel.margin = unit(c(0, 0, 0, 0), "null"),
        axis.ticks = element_blank()) +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)]) +
  scale_alpha(guide="none") +
  geom_quasirandom(dodge.width = 0.000001, color = "black")

fig8d <- ggplot(data=family_data_gen, aes(x=family, y=rate, fill=family, color=family, alpha=0.6)) +
  geom_boxplot() +
  labs(x=NULL, y=NULL) +
  theme_light() +
  theme(text = element_text(size = 14)) +
  scale_fill_manual(values=pnw_palette(n=6,name="Cascades")[c(1,4,5)]) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(1,4,5)]) +
  scale_alpha(guide="none") +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0)), legend.position = "none") +
  geom_quasirandom(dodge.width = 1, color = "black") 

top <- wrap_elements((fig8a + inset_element(fig8ainset, 0.002, 0.61, 0.4, 0.99)) + (fig8c + inset_element(fig8cinset, 0.002, 0.002, 0.4, 0.35))) +
  labs(tag = "Infraorder") +
  theme(
    plot.tag = element_text(size = rel(1.3)),
    plot.tag.position = "bottom"
  )

bottom <- wrap_elements(fig8b + fig8d) +
  labs(tag = "Family") +
  theme(
    plot.tag = element_text(size = rel(1.3)),
    plot.tag.position = "bottom"
  )

left <- wrap_elements((fig8a + inset_element(fig8ainset, 0.002, 0.61, 0.4, 0.99))/ fig8b) +
  labs(tag = "Mutations/site/year") +
  theme(
    plot.tag = element_text(size = rel(1.3), angle = 90),
    plot.tag.position = "left"
  )

right <- wrap_elements((fig8c + inset_element(fig8cinset, 0.002, 0.002, 0.4, 0.35))/ fig8d) +
  labs(tag = "Mutations/site/generation") +
  theme(
    plot.tag = element_text(size = rel(1.3), angle = 90),
    plot.tag.position = "left"
  )

left | right
```

<br>

<br>

None of the lifespan, asymptotic length, or generation time were found to significantly correlate with annual mutation rate estimates (p > 0.05 for all terms) when using a generalized linear model approach that accounts for phylogenetic inertia. However, when considering generational mutation rate for lifespan and asymptotic length only we found that the AIC-selected model included a significant positive correlation with lifespan (p = `r round(lifespan_tTable[2,4], 4)`, Figure 9), such that generational mutation rate increases with increasing lifespan. Length was not a significant predictor of mutation rate when considered as the only term or in combination with lifespan.

<br>

<br>

```{r demographic_pgls, fig.align = "center", fig.cap = "Figure 9. Variability in generational mutation rate by species' asymptotic lifespan."}

ggplot(data = species_mrate_gen) +
  geom_point(aes(x = lifespan, y = rate, color = infraorder), size = 3) +
  theme_light() +
  geom_abline(slope = coef(lifespan_pgls_gen)[2],
              intercept = coef(lifespan_pgls_gen)[1]) +
  geom_ribbon(data = ci, aes(x= x, ymin = ymin, 
                                  ymax = ymax), alpha = 0.3, fill = "grey50")+
  xlim(20,150)+
  ylim(0,1e-8) +
  labs(x="Lifespan", y="Mutations/site/generation") +
  theme_light() +
  theme(text = element_text(size = 16)) +
  theme(plot.margin = unit(c(10,30,0,0), 'pt'), axis.title.y = element_text(margin = margin(t=0,r=12,b=0,l=5)),
        axis.title.x = element_text(margin = margin(t=12,r=0,b=5,l=0))) +
  scale_color_manual(values=pnw_palette(n=6,name="Cascades")[c(3,6)],
                     name = "Infraorder")

```

<br>

<br>

## Discussion {-}

Mutation rate variability was high among species, supporting the use of species-specific mutation rates for phylogenetic analyses (REFS). Among pipeline effects on mutation rate, the choice of reference genome had the greatest effect on species-specific estimates of whole genome mutation rate, and the distance of the reference genome made up a large portion of this variability. Variability caused by reference genome was followed in effect size by the difference in mutation rate caused by corrections for ancestral heterozygosity and in coding vs. non-coding regions. Variation in divergence date estimates had comparatively minimal effect on mutation rate variability; this may be caused by the overall precision among the three sources used for divergence date estimates. 

The reference-driven differences in mutation rate observed in this study may be caused by several factors. For example, it is possible that the sperm whale evolutionary lineage has an overall lower mutation rate than other evolutionary lineages used as reference in this study. It is also possible that mutation rate accelerates and slows over time, and that mutation rates were higher in deep evolutionary time, slowed in mid-evolutionary time, and are marginally higher in shallow or contemporary evolutionary time. By comparing this pattern in mysticetes and odontocetes, we observed that the annual mutation rates of these two evolutionary lineages was not significantly different over deep evolutionary time, but differed significantly over intermediate and recent evolutionary time (Figure 2), suggesting that the annual mutation rates of each of these families may have embarked on differing evolutionary trajectories once the two families split from each other, resulting in the significantly lower annual mutation rates observed in contemporary mysticetes than in contemporary odontocetes (Figure 8a). However, our results indicate that, even with lower annual mutation rates, mysticetes have generational mutation rates that are slightly (but not significantly) larger than odotocetes (Figure 8c). This pattern can largely be explained by the increased lifespan of mysticetes, which significantly increases generational mutation rate (Figure 9) even at significantly lower annual mutation rates.

The degree of variability driven by choice of reference genome - up to an order of magnitude in some cetacean species - brings with it the question of how to choose the correct reference genome for alignment of new genomic data. Because mutation rate estimates represent an average over the evolutionary time period since divergence from the reference species it may be best to choose a reference genome that reflects the time period of the question. For example, population genomics or other studies concerned with recent evolutionary time might choose a reference genome from a recently-diverged species, and estimate mutation rate based on the reference species for use in downstream analyses, e.g. PSMC or BEAST phylogenies.

Correcting for species-specific ancestral heterozygosity has a linearly predictable effect on mutation rate, because $/pi_{anc}$ is simply subtracted from estimated genomic divergence before dividing by divergence time. We expect that subtracting any value from genomic divergence will cause a significant effect on the estimated mutation rate; the magnitude of the effect is determined by the size of $/pi_{anc}$. Our dataset confirmed this expectation. Further, we found that correcting for species-specific genomic heterozygosity was comparable with mean and minimum contemporary cetacean genomic heterozygosity; correcting for heterozygosity in this way has a minimal, but significant, effect on mutation rate estimates. However, our estimates of contemporary cetacean genomic heterozygosity ranged by over an order of magnitude (min = `r het_summary$min_pi`, max = `r het_summary$max_pi`), and correcting for ancestral heterozygosity using larger values had a much larger effect on mutation rate estimates. Because species-specific ancestral heterozygosity is currently unknowable, it is important to carefully consider whether and how to account for ancestral heterozygosity when estimating species-specific mutation rates, and to clearly describe this correction in the methods of genomic studies.

The minimal effect of divergence date estimates on downstream mutation rates indicates a high level of precision and convergence among our current estimates of divergence dates among cetacean species.

Compared to whole-genome mutation rate estimates among species, estimates of mutation rates among chromosomes varied less. Chromosome XX was the exception to this, likely driven at least in part by XXXX.

Species-specific mutation rates evolve over time, and are affected by demographic parameters like lifespan, generation time, and population size. Pedigree-based mutation rate estimates are a promising emerging tool that produce highly accurate estimates of mutation rate via counts of de novo mutations in parent-offspring trios. However, important to keep in mind that these highly accurate estimates of mutation rate will reflect contemporary population demographics and may differ significantly from historical mutation rates. Our results also indicate that variability in generational and annual mutation rates are not linearly correlated, but are linked via a species' generation time. **More here, REFS from Suarez-Menendes et al and Bergeron et al, Sung et al.).** 

## REFERENCES {-#REFERENCES}
<div id="refs"></div>

<br>

<br>

